packages/react-components/react-table/stories/DataGrid/Virtualization.stories.tsx

```diff
@@ -0,0 +1,195 @@
+        },
+      }),
+      createColumn<Item>({
+        columnId: 'author',
+        compare: (a, b) => {
+          return a.author.label.localeCompare(b.author.label);
+        },
+        renderHeaderCell: () => {
+          return 'Author';
+        },
+        renderCell: item => {
+          return (
+            <TableCellLayout media={<Avatar badge={{ status: item.author.status }} />}>
+              {item.author.label}
+            </TableCellLayout>
+          );
+        },
+      }),
+      createColumn<Item>({
+        columnId: 'lastUpdated',
+        compare: (a, b) => {
+          return a.lastUpdated.timestamp - b.lastUpdated.timestamp;
+        },
+        renderHeaderCell: () => {
+          return 'Last updated';
+        },
+
+        renderCell: item => {
+          return item.lastUpdated.label;
+        },
+      }),
+      createColumn<Item>({
+        columnId: 'lastUpdate',
+        compare: (a, b) => {
+          return a.lastUpdate.label.localeCompare(b.lastUpdate.label);
+        },
+        renderHeaderCell: () => {
+          return 'Last update';
+        },
+        renderCell: item => {
+          return <TableCellLayout media={item.lastUpdate.icon}>{item.lastUpdate.label}</TableCellLayout>;
+        },
+      }),
+    ],
+    [],
+  );
+
+  const items = React.useMemo(() => {
+    const baseItems: Item[] = [
+      {
+        file: { label: 'Meeting notes', icon: <DocumentRegular /> },
+        author: { label: 'Max Mustermann', status: 'available' },
+        lastUpdated: { label: '7h ago', timestamp: 1 },
+        lastUpdate: {
+          label: 'You edited this',
+          icon: <EditRegular />,
+        },
+      },
+      {
+        file: { label: 'Thursday presentation', icon: <FolderRegular /> },
+        author: { label: 'Erika Mustermann', status: 'busy' },
+        lastUpdated: { label: 'Yesterday at 1:45 PM', timestamp: 2 },
+        lastUpdate: {
+          label: 'You recently opened this',
+          icon: <OpenRegular />,
+        },
+      },
+      {
+        file: { label: 'Training recording', icon: <VideoRegular /> },
+        author: { label: 'John Doe', status: 'away' },
+        lastUpdated: { label: 'Yesterday at 1:45 PM', timestamp: 2 },
+        lastUpdate: {
+          label: 'You recently opened this',
+          icon: <OpenRegular />,
+        },
+      },
+      {
+        file: { label: 'Purchase order', icon: <DocumentPdfRegular /> },
+        author: { label: 'Jane Doe', status: 'offline' },
+        lastUpdated: { label: 'Tue at 9:30 AM', timestamp: 3 },
+        lastUpdate: {
+          label: 'You shared this in a Teams chat',
+          icon: <PeopleRegular />,
+        },
+      },
+    ];
+
+    return new Array(1500).fill(0).map((_, i) => baseItems[i % baseItems.length]);
+  }, []);
+
+  return (
+    <DataGrid items={items} columns={columns} focusMode="cell" sortable selectionMode="multiselect">
+      <DataGridHeader style={{ paddingRight: scrollbarWidth }}>
```

Comment:
This is a bit worrying TBH. What are expectations on consumer side? Will they copy-paste `useScrollbarWidth()` hook?

Reply:
Updated with the now official `useScrollbarWidth` utility implemented as in #26144